content/development/testing.md

```diff
@@ -9,7 +9,312 @@ chapter: false
 
 Well, you managed to write your rule, but now want to see if if can be added to the CRS? This document should help you to test it using the same tooling the project uses for its tests.
 
-Tests are performed using a Python tool called [ftw](https://github.com/coreruleset/ftw). We run them using a [GitHub actions pipeline](https://github.com/coreruleset/coreruleset/blob/{{< param crs_dev_branch >}}/dev/.github/workflows/test.yml). You can easily reproduce that locally, in your workstation.
+CRS uses [go-ftw](https://github.com/coreruleset/go-ftw) to run test cases. **go-ftw** is the successor of the previously used test runner [ftw](https://github.com/coreruleset/ftw). The CRS project no longer uses **ftw** but it us still useful for running tests of older CRS versions.
+
+## Environments
+
+Before you start to run tests, you should set up your environment. You can use Docker to run a web server with CRS integration or use your existing environment.
+
+### Setting up Docker containers
+
+For testing, we use the [container images from our project](https://github.com/coreruleset/modsecurity-crs-docker). We "bind mount" the rules in the CRS Git repository to the web server container and then instruct **go-ftw** to send requests to it.
+
+To test we need two containers: the WAF itself, and a backend, provided in this case by the http://httpbin.org project. The `docker-compose.yml` in the CRS Git repository is a ready-to-run configuration for testing, to be used with the `docker compose` command.
+
+-> The supported platform is modsecurity 2 with Apache
+
+Let's start the containers using:
+```bash
+❯ docker compose -f tests/docker-compose.yml up -d modsec2-apache
+
+Creating network "tests_default" with the default driver
+Creating tests_backend_1 ... done
+Creating modsec2-apache  ... done
+❯ docker ps
+CONTAINER ID   IMAGE                              COMMAND                  CREATED          STATUS         PORTS                               NAMES
+785a6f6a3cb6   owasp/modsecurity-crs:3.3-apache   "/docker-entrypoint.…"   7 seconds ago    Up 3 seconds   0.0.0.0:80->80/tcp, :::80->80/tcp   modsec2-apache
+bb8d5a7f256d   kennethreitz/httpbin               "gunicorn -b 0.0.0.0…"   10 seconds ago   Up 7 seconds   80/tcp                              tests_backend_1
+```
+
+Excellent, our containers are running, now we can start our tests.
+
+### Using your own environment for testing
+
+If you have your own environment set up, you can configure that for testing. If you don't yet have an environment but have a Debian or Ubuntu server, you can configure [Digitalwave's](https://modsecurity.digitalwave.hu) package repository for supported distributions. There you can follow the instructions on how to configure your WAF.
```

Comment:
```suggestion
If you have your own environment set up, you can configure that for testing. Please [follow these instructions]({{< ref "install.md#installing-a-compatible-waf-engine" >}}) to install the WAF server locally.
```

Reply:
I don't think it makes sense to repeat that here, let's point it to the page we have it documented...